### PR TITLE
fix: make JS files take precedence over TS files with the same name

### PR DESCRIPTION
**Pull request by eduardoboucas**
**- Summary**

When we started introducing support for TS files, the goal was for them to take precedence over JS functions with the same name. This was the case when the functions were in a sub-directory (e.g. `functions/sub-dir/function.js` vs. `functions/sub-dir/function.ts`). Unfortunately, this wasn't the case when the functions were at the root level (e.g. `functions/function.js` vs `functions/function.ts`). This PR addresses that.

**- Test plan**

- Added test to assert that `.js` files take precedence over `.ts` files
- Added test to assert that  `.js` files take precedence over `.zip` files (I verified this was the current behaviour) before we introduced these changes

**- Description for the changelog**

fix: make JS files take precedence over TS files with the same name

**- A picture of a cute animal (not mandatory but encouraged)**

images

### CODE REVIEW

File: tests/main.js [hunk 1]
```diff
@@ -402,11 +402,11 @@ testBundlers(
     t.deepEqual(
       functions,
       [
-        { name: 'test', mainFile: 'test.js', runtime: 'js', extension: '.js' },
```
Comment (eduardoboucas) [inline tests/main.js:405]: This changes the order of these files once again. 😞

File: tests/main.js [hunk 2]
```diff
@@ -402,8 +402,8 @@ testBundlers(
...
         { name: 'test', mainFile: 'test.zip', runtime: 'js', extension: '.zip' },
+        { name: 'test', mainFile: 'test.js', runtime: 'js', extension: '.js' },
```
Comment (eduardoboucas) [inline tests/main.js:406]: The reason for this changing is that we're now prioritising `.js` files over `.zip` files, which was the existing behaviour before #364.

File: src/runtimes/node/finder.js [hunk 3]
```diff
@@ -6,19 +6,42 @@ const locatePath = require('locate-path')
...
   // eslint-disable-next-line fp/no-mutating-methods
-  const sortedFunctions = functions
-    .filter(Boolean)
-    .sort((fA, fB) => Number(fA.stat.isDirectory()) - Number(fB.stat.isDirectory()))
+  const sortedFunctions = functions.filter(Boolean).sort((fA, fB) => {
+    // We first sort the functions array to put directories first. This is so
+    // that `{name}/{name}.js` takes precedence over `{name}.js`.
+    const directorySort = Number(fA.stat.isDirectory()) - Number(fB.stat.isDirectory())
+
+    if (directorySort) {
```
Comment (ehmicky) [inline src/runtimes/node/finder.js:33]:
```suggestion
    if (directorySort !== 0) {
```

Should we explicit about the typecasting?

**Review by ehmicky — APPROVED** [review]

File: src/runtimes/node/finder.js (hunk 3 above)
Comment (eduardoboucas) [reply src/runtimes/node/finder.js]: Done in https://github.com/netlify/zip-it-and-ship-it/pull/374/commits/4e6e530a5e1d417baf4a566e52f89d87899b5c05.

Comment (eduardoboucas) [reply src/runtimes/node/finder.js]: (Oh, just realised you made an actual suggestion which I could've committed directly!)

**Pull request merged**